src/plugins/blackbox/public/js/blackbox.js

```diff
@@ -22,26 +22,36 @@
   head.appendChild(js);
   var Blackbox = function Blackbox(cockpit) {
     console.log('Loading Blackbox plugin.');
+    var self = this;
+
     this.cockpit = cockpit;
     this.recording = false;
     this.idb;
     this.sessionID = this.newSession();
     this.eventBuffer = [];
     this.otherBuffer = [];
     this.sessions_cache = [];
-  };
-  plugins.Blackbox = Blackbox;
-  Blackbox.prototype.inputDefaults = function inputDefaults() {
-    var self = this;
-    return [{
-        name: 'blackbox.record',
-        description: 'Start recording telemetry data.',
-        defaults: { keyboard: 'r' },
-        down: function () {
+
+    this.inputDefaults = [
+    {
+      name: 'blackbox.record',
+      shortName: "Toggle Blackbox recording",
+      description: 'Start recording the telemetry data.',
+      controllers: new Map([["keyboard", "r"]]),
+      actions:
+      {
+        down: function() {
           self.toggleRecording();
         }
-      }];
+      }
+    }];
+
+    //If this was loaded after the input manager, let it know we are ready to be loaded
```

Comment:
This should not be needed. Unless the input manager changed, it actually crawls the actual array of loaded plugins and directly calls the inputDefaults method rather than trying to invoke them via the bus.
I see the pattern repeated all over. You should be able to delete all of them.